src/main/java/com/unb/tracker/web/CourseController.java

```diff
@@ -105,14 +111,72 @@ public String postCourse(@RequestBody Course course) {
 
     @PostMapping("/course")
     public String courseSave(@ModelAttribute Course course, ModelMap map, Principal principal) {
-        LOG.info("courseSave - starting - principle.name: {}", principal.getName());
         User user = userRepository.findByUsername(principal.getName());
-        if (user == null || !user.getHasExtendedPrivileges()) {
+        if (user == null || !user.getHasExtendedPrivileges())
             throw new BadRequestException();
+
+        //If user wishes to use grid from other course
+        Long courseGridReuseID = course.getCourseGridReuseID();
+        if(courseGridReuseID != null)
+        {
+            Course otherCourse = courseRepository.findOne(courseGridReuseID);
+            reuseCourseGridHelper(course, otherCourse);
         }
+
         course.setInstructor(user);
         courseRepository.save(course);
         map.addAttribute("course", course);
         return "redirect:/" + user.getUsername() + "/" + course.getName() + "/" + course.getSection();
     }
+
+    @GetMapping("/courses/query/{queryString}")
+    public @ResponseBody List<Course> queryCourses(@PathVariable String queryString) {
+        return courseRepository.findByPartialName(queryString);
+    }
+
+    @PostMapping("/course/gridReuse")
+    public @ResponseBody Course reuseCourseGrid(@RequestBody Map<String, Long> courseIds, Principal principal) throws BadRequestException {
+        User user = userRepository.findByUsername(principal.getName());
+        if (user == null || !user.getHasExtendedPrivileges())
+            throw new BadRequestException();
+
+        Long courseReceiveID = courseIds.get("currentCourse");
```

Comment:
I'm a bit confused by the variable names here, which course is `courseGive` and which is `courseReceive`?

Reply:
I actually had a hard time trying to come up with good variable names here haha. `courseGive` GIVES their seat plan to `courseReceive`. I know, the names aren't ideal, but im definitely up for suggestions!